src/chat.rs

```diff
@@ -3338,9 +3338,36 @@ pub async fn marknoticed_chat(context: &Context, chat_id: ChatId) -> Result<()>
             AND chat_id=?;",
                 (MessageState::InNoticed, MessageState::InFresh, chat_id),
             )
-            .await?
-            == 0
-        {
+            .await?;
+
+        // This is to trigger emitting `MsgsNoticed` on other devices when reactions are noticed
+        // locally (i.e. when the chat was opened locally).
+        let hidden_messages = context
+            .sql
+            .query_map(
+                "SELECT id, rfc724_mid FROM msgs
+                    WHERE state=?
+                      AND hidden=1
+                      AND chat_id=?
+                    ORDER BY id DESC LIMIT 100", // LIMIT to 100 in order to avoid blocking the UI too long, usually there will be less than 100 messages anyway
+                (MessageState::InFresh, chat_id), // No need to check for InNoticed messages, because reactions are never InNoticed
+                |row| {
+                    let msg_id: MsgId = row.get(0)?;
+                    let rfc724_mid: String = row.get(1)?;
+                    Ok((msg_id, rfc724_mid))
+                },
+                |rows| {
+                    rows.collect::<std::result::Result<Vec<_>, _>>()
+                        .map_err(Into::into)
+                },
+            )
+            .await?;
+        for (msg_id, rfc724_mid) in &hidden_messages {
```

Comment:
Do we really need to mark all hidden messages as seen on IMAP? In #6213 i only took the last hidden message and checked if it needs to be marked as seen, it seems this is enough to trigger `MsgsNoticed` on other devices and should result in better performance

Reply:
Maybe then for now remove `DESC` from `ORDER BY id DESC LIMIT 100` above so that reactions are processed in some well-defined order? There shouldn't be more than 100 reactions usually, but still